sparkle/src/components/FeedbackSelector.tsx

```diff
@@ -49,26 +46,92 @@ export function FeedbackSelector({
   const [popOverInfo, setPopoverInfo] = React.useState<JSX.Element | null>(
     null
   );
+  const [isConversationShared, setIsConversationShared] = React.useState(
+    feedback?.isConversationShared ?? false
+  );
+  // This is required to adjust the content of the popover even when feedback is null.
+  const [lastSelectedThumb, setLastSelectedThumb] =
+    React.useState<ThumbReaction | null>(feedback?.thumb ?? null);
 
   useEffect(() => {
     if (isPopoverOpen) {
       if (getPopoverInfo) {
         setPopoverInfo(getPopoverInfo());
       }
-      setLocalFeedbackContent(feedback?.feedbackContent ?? null);
+      if (feedback?.thumb === lastSelectedThumb) {
+        setLocalFeedbackContent(feedback?.feedbackContent ?? null);
+      }
     }
-  }, [isPopoverOpen, feedback?.feedbackContent]);
+  }, [
+    isPopoverOpen,
+    feedback?.feedbackContent,
+    getPopoverInfo,
+    lastSelectedThumb,
+  ]);
+
+  const selectThumb = useCallback(
+    async (thumb: ThumbReaction) => {
+      // Whether to remove the thumb reaction
+      const isToRemove = feedback?.thumb === thumb;
+      setIsPopoverOpen(!isToRemove);
+      setLastSelectedThumb(isToRemove ? null : thumb);
 
-  const selectThumb = async (thumb: ThumbReaction) => {
-    const isToRemove = feedback?.thumb === thumb;
-    setIsPopoverOpen(!isToRemove);
+      // Checkbox ticked by default only for new thumbs down
+      setIsConversationShared(thumb === "down");
 
-    await onSubmitThumb({
-      feedbackContent: localFeedbackContent,
-      thumb,
-      isToRemove,
-    });
-  };
+      // We enforce written feedback for thumbs down.
+      // -> Not saving the reaction until then.
+      if (thumb === "down" && !isToRemove) {
+        return;
+      }
+
+      await onSubmitThumb({
+        feedbackContent: localFeedbackContent,
+        thumb,
+        isToRemove,
+        // The sharing option was never displayed so far -> Opt out of sharing.
+        isConversationShared: false,
+      });
+    },
+    [feedback?.thumb, localFeedbackContent, onSubmitThumb, isConversationShared]
+  );
+
+  const handleThumbUp = useCallback(async () => {
+    await selectThumb("up");
+  }, [selectThumb]);
+
+  const handleThumbDown = useCallback(async () => {
+    await selectThumb("down");
+  }, [selectThumb]);
+
+  const handleTextAreaChange = useCallback(
+    (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+      setLocalFeedbackContent(e.target.value);
+    },
+    []
+  );
+
+  const closePopover = useCallback(() => {
```

Comment:
No need to wrap setters for state

Reply:
This is true if we pass the set function as is, but to my understanding here we lso inject 'false', so it would trigger a rerender (because we'd pass a arrow function and not he function name)

Reply:
okok